Bump version to v1.10.0

tiborvass opened the pull request:
Signed-off-by: Tibor Vass tibor@docker.com

Commit:
Improvements to ANSI emulation in conemu

Signed-off-by: John Howard <jhoward@microsoft.com>

Commit:
Fix volume filter validation

Fixes: #18890
This fix add same filter validation logic as images. We should
add such check to make sure filters work make sense to end-users

Right now, we keep old use 1 as filter, but in long term, it should
be have same interface checking as images, it could be improved in
other patches.

Signed-off-by: Kai Qiang Wu(Kennan) <wkqwu@cn.ibm.com>

cyphar committed:
integration-cli: add bad --cgroup-parent tests

To ensure we don't regress on bad --cgroup-parent paths, add some
integration tests that check that the host hasn't toppled (or suddently
started to create files in the host).

Signed-off-by: Aleksa Sarai <asarai@suse.com>

arun-gupta committed:
adding docs for Dockerizing Couchbase service

cleaning up instructions, removing redundant output

Signed-off-by: Arun Gupta <arun.gupta@gmail.com>

adding the latest feedback

Signed-off-by: Arun Gupta <arun.gupta@gmail.com>

mikedougherty committed:
Adjust version string processing for RPM build

This will only trim off the last '-' separated token to be checked for
RC status. This allows including a 'cs#' token in the version string.

Signed-off-by: Mike Dougherty <mike.dougherty@docker.com>

Djelibeybi committed:
Restoring the RPM build process for Oracle Linux 6 and updating the docker-engine.spec
file to require the Unbreakable Enterprise Kernel Release 4 on both Oracle Linux 6
and Oracle Linux 7.

The UEK R4 provides the required kernel functionality for VxLAN support
required by Docker 1.9 and user namespace support required for 1.10+.

The build of Docker on Oracle Linux 6 requires some manipulation of the build
environment so that the CGO compiler uses the UEK R4 headers instead of the old
default kernel headers.

Signed-off-by: Avi Miller <avi.miller@oracle.com>

Djelibeybi committed:
Add missing ? to specfile variable.

Signed-off-by: Avi Miller <avi.miller@oracle.com>

Commit:
Allow external volume drivers to host anonymous volumes and copy existing data from image.

Signed-off-by: Stephen Rust <srust@blockbridge.com>

scaleoutsean committed:
Add GPFS

Signed-off-by: Sean Lee <seanlee@tw.ibm.com>

nigelpoulton committed:
Updating for CAS changes and new select a driver section
Adding changes requested by @jamtur01 and wrapping to 80 chars
Adding typo fixes and replacing tabs with 4xspaces
Closes and fixes #19240
Updating with content addressable storage model changes

Signed-off-by: Nigel <nigelpoulton@hotmail.com>

karkhaz committed:
Added `dead` to docs for docker ps -f status=...

It is possible to invoke `docker ps -f status=dead`, but the
documentation for docker-ps does not mention `dead` as a valid option.
This commit fixes that.

Signed-off-by: Kareem Khazem <karkhaz@karkhaz.com>

jessfraz committed:
add warning if upgrading via script to migrate

Signed-off-by: Jessica Frazelle <acidburn@docker.com>

Commit:
Bump version to v1.10.0

Signed-off-by: Tibor Vass <tibor@docker.com>

dqminh committed:
only close LogDriver after LogCopier is done

this prevents the copier from sending messages in the buffer to the closed
driver. If the copied took longer than the timeout to drain the buffer, this
aborts the copier read loop and return back so we can cleanup resources
properly.

Signed-off-by: Daniel Dao <dqminh@cloudflare.com>

cpuguy83 committed:
Fix panic on starting exec more than once

Issue was caused when exec is tarted, exits, then stated again.
In this case, `Close` is called twice, which closes a channel twice.

Changes execConfig.ExitCode to a pointer so we can test if the it has
been set or not.
This allows us to return early when the exec has already been run.

Signed-off-by: Brian Goff <cpuguy83@gmail.com>

thaJeztah commented on CHANGELOG.md:
```diff
@@ -5,6 +5,116 @@ information on the list of deprecated flags and APIs please have a look at
 https://docs.docker.com/misc/deprecated/ where target removal dates can also
 be found.
 
+## 1.10 (2016-02-04)
```
1.10.0 ?

vdemeester commented on CHANGELOG.md:
```diff
@@ -5,6 +5,116 @@ information on the list of deprecated flags and APIs please have a look at
++ Add `--device-read-iops` and `--device-write-iops` flags to `docker run` [#15879](https://github.com/docker/docker/pull/15879)
++ Add `--oom-score-adj` flag to `docker run` [#16277](https://github.com/docker/docker/pull/16277)
+* Change `docker run` exit codes to distinguish between runtime and application errors [#14012](https://github.com/docker/docker/pull/14012)
++ Add `--detach-keys` flag to `attach` and `exec` commands to override the default key sequence that detaches from a container  [#15666](https://github.com/docker/docker/pull/15666)
```
to `run` and `start` too :stuck_out_tongue_closed_eyes:

tiborvass replied:
argh too late, we'll fix for final :) thanks

icecrime commented:
We have just pushed 1.10.0-rc1 to test.docker.com:

> Ubuntu/Debian: curl -fsSL https://test.docker.com/ | sh  
> Linux 64bit binary: https://test.docker.com/builds/Linux/x86_64/docker-1.10.0-rc1  
> Darwin/OSX 64bit client binary: https://test.docker.com/builds/Darwin/x86_64/docker-1.10.0-rc1  
> Linux 64bit tgz: https://test.docker.com/builds/Linux/x86_64/docker-1.10.0-rc1.tgz  
> Windows 64bit client binary: https://test.docker.com/builds/Windows/x86_64/docker-1.10.0-rc1.exe  
> Windows 32bit client binary: https://test.docker.com/builds/Windows/i386/docker-1.10.0-rc1.exe

tianon commented:
boot2docker ISO is over at https://github.com/tianon/boot2docker-legacy/releases/tag/v1.10.0-rc1

jessfraz committed:
fix opensuse rpm

Signed-off-by: Jessica Frazelle <acidburn@docker.com>

calavera committed:
Extract container store from the daemon.

- Generalize in an interface.
- Stop abusing of List for everything.

Signed-off-by: David Calavera <david.calavera@gmail.com>

vikstrous commented on CHANGELOG.md:
```diff
@@ -5,6 +5,116 @@ information on the list of deprecated flags and APIs please have a look at
+
+### Distribution
+
++ New `registrytoken` pass-thru token type for `AuthConfig` [#17481](https://github.com/docker/docker/pull/17481)  
```
I'm pretty sure this is supposed to be #17741

tiborvass replied:
Indeed, my bad thanks for catching it

thaJeztah commented:
Found one more feature that we didn't mention in the changelog; https://github.com/docker/docker/pull/15780

runcom committed:
api: client: build: do not fall through if git isn't installed

Signed-off-by: Antonio Murdaca <runcom@redhat.com>

sdurrheimer committed:
Add zsh completion for 'docker network create --internal'

Signed-off-by: Steve Durrheimer <s.durrheimer@gmail.com>

sdurrheimer committed:
Modify zsh completion for connecting/disconnecting non-running containers to networks

Signed-off-by: Steve Durrheimer <s.durrheimer@gmail.com>

cdauth committed:
Set TasksMax in addition to LimitNPROC in systemd service files

systemd sets an additional limit on processes and threads that defaults to 512 when run under Linux >= 4.3.
See more information here: http://unix.stackexchange.com/a/255603/59955

Signed-off-by: Candid Dauth <cdauth@cdauth.eu>

Commit:
Vendoring miekg/dns @ 75e6e86cc601825c5dbcd4e0c209eab180997cd7

- Fixes the issue of Shutdown() not working, resulting in hung
  goroutines

Signed-off-by: Santhosh Manohar <santhosh@docker.com>

mavenugo committed:
IT for remote network driver & ungraceful restart

Signed-off-by: Madhu Venugopal <madhu@docker.com>

Commit:
docs: document options for default network driver

Fixes issue #18410

Signed-off-by: Wen Cheng Ma <wenchma@cn.ibm.com>

albers committed:
bash completion for container linking and aliasing

Signed-off-by: Harald Albers <github@albersweb.de>

albers committed:
bash completion for `docker network create --internal, --ipam-opt`

Signed-off-by: Harald Albers <github@albersweb.de>

albers committed:
Support new events in bash completion

Signed-off-by: Harald Albers <github@albersweb.de>

albers committed:
Refactor handling of key specific subcompletions

The currently used idiom for handling key specific subcompletions
did not work here: behind `docker event -f type=network `, the completion
of networks triggered. The expected behaviour is not to complete
anything here.

In order to limit the scope of the corresponding PR, the new idiom is
currently only used in `docker events --filter`.

Signed-off-by: Harald Albers <github@albersweb.de>

clnperez committed:
Move closeNotify to fix panic with newer golang

This is happening now due to improvements in net/http:
https://github.com/golang/go/commit/99fb19194c03c618c0d8faa87b91ba419ae28ee3

To test, change the go version in the Dockerfile:
-ENV GO_VERSION 1.5.3
+ENV GO_VERSION 1.6beta2

More info here: https://github.com/golang/go/issues/14001

Signed-off-by: Christy Perez <christy@linux.vnet.ibm.com>

jessfraz committed:
disable install of docker-engine for 1.10, try to get ready for 1.11

Signed-off-by: Jessica Frazelle <acidburn@docker.com>

dongluochen commented on CHANGELOG.md:
```diff
@@ -5,6 +5,116 @@ information on the list of deprecated flags and APIs please have a look at
+  `docker load`/`docker save` tarballs now also contain content-addressable image configurations.
+* Add support for the new [manifest format ("schema2")](https://github.com/docker/distribution/blob/master/docs/spec/manifest-v2-2.md) [#18785](https://github.com/docker/docker/pull/18785)
+* Lots of improvements for push and pull: performance++, retries on failed downloads, cancelling on client disconnect [#18353](https://github.com/docker/docker/pull/18353), [#18418](https://github.com/docker/docker/pull/18418), [#19109](https://github.com/docker/docker/pull/19109), [#18353](https://github.com/docker/docker/pull/18353)
+* Limit v1 protocol fallbacks [#18590](https://github.cAom/docker/docker/pull/18590)
```
Broken link `github.cAom`.

tiborvass replied:
wow... thanks sorry about that

Commit:
Revert reporting of multiple pull errors

Revert the portions of #17617 that report all errors when a pull
falls back, and go back to just reporting the last error. This was nice
to have, but causes some UX issues because nonexistent images show
additional "unauthorized" errors.

Keep the part of the PR that handled ENOSPC, as this appears to work
even without tracking multiple errors.

Fixes #19419

Signed-off-by: Aaron Lehmann <aaron.lehmann@docker.com>

thaJeztah commented:
For the next RC, perhaps we should add some information on stability-improvements that were made to libnetwork (among which, corruption/incorrect lib-kv.db), and improvements when using in a multi-node setup https://github.com/docker/docker/issues/18527#issuecomment-172671287, https://github.com/docker/docker/issues/18870#issuecomment-172673982

mavenugo committed:
nil ptr check for endpointsettings when used with older clients

Signed-off-by: Madhu Venugopal <madhu@docker.com>

WeiZhang555 committed:
bugfix: prevent creating network 'default'

Default is predefined network and is reserved, so we should stop user
from creating network with name `default`

Signed-off-by: Zhang Wei <zhangwei555@huawei.com>

jessfraz committed:
add send, recv, and x32 so we can install i386 pkgs on amd64

Signed-off-by: Jessica Frazelle <acidburn@docker.com>

Commit:
Fix the privileged example

Signed-off-by: Kai Qiang Wu(Kennan) <wkqwu@cn.ibm.com>

Commit:
Refine the volume mount example

The path here should be absolute, else it would
deem it as volume name.

Also link to release page to contain static binary,
the old link not work, because it is just used to
install docker in os distro, it can not be used
as static binary directly.

Signed-off-by: Kai Qiang Wu(Kennan) <wkqwu@cn.ibm.com>

Commit:
Fix add host device example

The example is not right in parameter, and also
one command is same as first one, it should be typo
before, we should use 'rw' as example for that.

Signed-off-by: Kai Qiang Wu(Kennan) <wkqwu@cn.ibm.com>

Commit:
Fix ulimit command form

The ulimit is builtin, so we need shell form to execute it.

Signed-off-by: Kai Qiang Wu(Kennan) <wkqwu@cn.ibm.com>

patricksuo committed:
Fix race condition in execCommandGC

`daemon.execCommandGC`
The daemon object (grep execCommandGC) iterate over a map
(grep execCommands.Commands) in a goroutine.
Lock can't protect concurrency access in this case.
Exec command storage object should return a copy of commands instead.

Signed-off-by: Pei Su <sillyousu@gmail.com>

mortonfox committed:
cgroups documentation moved

Signed-off-by: Morton Fox <github@qslw.com>

calavera committed:
Fix post config verification without flags.

- Set the daemon log level to what's set in the configuration.
- Enable TLS when TLSVerify is enabled.

Signed-off-by: David Calavera <david.calavera@gmail.com>

dmcgowan committed:
Add more robust error handling on layer store creation

Add continue when layer fails on store creation
Trim whitespace from layerstore files to keep trailing space from failing a layer load

Fixes #19449

Signed-off-by: Derek McGowan <derek@mcgstyle.net> (github: dmcgowan)

sdurrheimer committed:
Add zsh completion for 'docker network create --ipam-opt' + Reordering for better maintainability

Signed-off-by: Steve Durrheimer <s.durrheimer@gmail.com>

sdurrheimer committed:
Add zsh completion for 'docker network connect --alias' and 'docker {create, run} --net-alias'

Signed-off-by: Steve Durrheimer <s.durrheimer@gmail.com>

sdurrheimer committed:
Add zsh completion for 'docker {network connect, create, run} --ip --ip6'

Signed-off-by: Steve Durrheimer <s.durrheimer@gmail.com>

sdurrheimer committed:
Add zsh completion for 'docker network connect --link'

Signed-off-by: Steve Durrheimer <s.durrheimer@gmail.com>

Commit:
Vendor updated docker/distribution package

Fixes #19400

Note that this introduces an incompatibility with Docker 1.10-rc1,
because the media type used for schema1 manifests has been corrected in
the upstream distribution code. Docker 1.10-rc1 won't be able to pull
old manifests from Registry 2.3-rc0 and up, but because of this vendor
update, Docker 1.10-rc2 won't have this problem.

Signed-off-by: Aaron Lehmann <aaron.lehmann@docker.com>

mavenugo committed:
Vendor libnetwork v0.6.0-rc1

- Fixes docker/docker#19404
- Fixes ungraceful daemon restart issue in systemd with remote
  network plugin (https://github.com/docker/libnetwork/issues/813)

Signed-off-by: Madhu Venugopal <madhu@docker.com>

sdurrheimer committed:
Add zsh completion for 'docker {attach,exec,run,start} --detach-keys'

Signed-off-by: Steve Durrheimer <s.durrheimer@gmail.com>

crosbymichael committed:
Update libcontainer to 47e3f834d73e76bc2a6a585b48d

This adds a fix for the resource struct in the cgroups type and seccomp
IsEnabled function

Signed-off-by: Michael Crosby <crosbymichael@gmail.com>

zchee committed:
Add `ARG` instruction syntax for vim

Signed-off-by: Koichi Shiraishi <k@zchee.io>

ywy1ywy committed:
fix a spelling error of 'dnsmasq'

Signed-off-by: Wenyu You <21551128@zju.edu.cn>

Commit:
Fix commit wrong repository example

The old name is invalid in new repository name spec.
So we need to fix them.

Signed-off-by: Kai Qiang Wu(Kennan) <wkqwu@cn.ibm.com>

Commit:
Fix minor doc grammar issues
Signed-off-by: Dave Barboza <dxbarboza@gmail.com>

cpuguy83 committed:
Fix loading of containerized plugins

During daemon startup, all containers are registered before any are
started.
During container registration it was calling out to initialize volumes.
If the volume uses a plugin that is running in a container, this will
cause the restart of that container to fail since the plugin is not yet
running.
This also slowed down daemon startup since volume initialization was
happening sequentially, which can be slow (and is flat out slow since
initialization would fail but take 8 seconds for each volume to do it).

This fix holds off on volume initialization until after containers are
restarted and does the initialization in parallel.

The containers that are restarted will have thier volumes initialized
because they are being started. If any of these containers are using a
plugin they will just keep retrying to reach the plugin (up to the
timeout, which is 8seconds) until the container with the plugin is up
and running.

Signed-off-by: Brian Goff <cpuguy83@gmail.com>

cpuguy83 committed:
Use fine-grained locks for plugin loading.

This helps ensure that only one thing is trying to intialize a plugin at
once while also keeping the global lock free during initialization.

Signed-off-by: Brian Goff <cpuguy83@gmail.com>

Commit:
Clarify error message when a .cert file is missing a corresponding key

The daemon uses two similar filename extensions to identify different
kinds of certificates. ".crt" files are interpreted as CA certificates,
and ".cert" files are interprted as client certificates. If a CA
certificate is accidentally given the extension ".cert", it will lead to
the following error message:

    Missing key ca.key for certificate ca.cert

To make this slightly less confusing, clarify the error message with a
note that CA certificates should use the extension ".crt".

Signed-off-by: Aaron Lehmann <aaron.lehmann@docker.com>

BrianBland committed:
Fixes layer MediaTypes in manifests created from a cross-repository push

Signed-off-by: Brian Bland <brian.bland@docker.com>

Commit:
Fixing missing certs article; consolidating security material
Entering comments from reviewers
Updating with Derek's comments
Fixing bad links reported by build

Signed-off-by: Mary Anthony <mary@docker.com>

crosbymichael committed:
Move tty set and restore to caller

Fixes #19506

This fixes the issue of errors on create and the tty not being able to
be restored to its previous state because of a race where it was
in the hijack goroutine.

Signed-off-by: Michael Crosby <crosbymichael@gmail.com>

cpuguy83 committed:
Don't error out when link name in use.

This preserves old behavior from sqlite links/names.

Signed-off-by: Brian Goff <cpuguy83@gmail.com>

calavera committed:
Verify that the configuration keys in the file are valid.

- Return an error if any of the keys don't match valid flags.
- Fix an issue ignoring merged values as named values.
- Fix tlsverify configuration key.
- Fix bug in mflag to avoid panics when one of the flag set doesn't have any flag.

Signed-off-by: David Calavera <david.calavera@gmail.com>

dmcgowan committed:
Unwrap URL errors on retry

When authorization errors are returned by the token process the error will be wrapped in url.Error.
In order to check the underlying error for retry this error message should be unwrapped.
Unwrapping this error allows failure to push due to an unauthorized response to keep from retrying, possibly resulting in later 429 responses.

Signed-off-by: Derek McGowan <derek@mcgstyle.net> (github: dmcgowan)

Commit:
Vendor updated distribution

The only changes are https://github.com/docker/distribution/pull/1379
and https://github.com/docker/distribution/pull/1380.

Fixes #19476

Signed-off-by: Aaron Lehmann <aaron.lehmann@docker.com>

coolljt0725 committed:
Fix #19477, clean up the ports when release network

Signed-off-by: Lei Jitang <leijitang@huawei.com>

aboch committed:
Move ErrUnsupportedNetwork* checks to updateNetworkConfig() func

Signed-off-by: Alessandro Boch <aboch@docker.com>

cpuguy83 committed:
Add back compat for volume drivers `Get` and `Ls`

Use a back-compat struct to handle listing volumes for volumes we know
about (because, presumably, they are being used by a container) for
volume drivers which don't yet support `List`.

Adds a fall-back for the volume driver `Get` call, which will use
`Create` when the driver returns a `404` for `Get`. The old behavior was
to always use `Create` to get a volume reference.

Signed-off-by: Brian Goff <cpuguy83@gmail.com>

Commit:
Fix the typo in ps

Signed-off-by: Kai Qiang Wu(Kennan) <wkqwu@cn.ibm.com>

hqhq committed:
Fix comment about swap limit of docker update

The description "set `-1` to disable swap" is wrong, `build`,
`create` and `run` already fixed, we need to fix `update` as well.

Signed-off-by: Qiang Huang <h.huangqiang@huawei.com>

Commit:
Fixed typo in "/etc/subUid" 

Signed-off-by: Azat Khuziyakhmetov <shadow_uz@mail.ru>

tonistiigi committed:
Revert "Copy aufs hardlinks to top layer"

This reverts commit ef05b83417e32f269daf798307adcf07fe6ef13f.

Signed-off-by: Tonis Tiigi <tonistiigi@gmail.com>

cpuguy83 committed:
Bump plugin API version

Signed-off-by: Brian Goff <cpuguy83@gmail.com>

Commit:
Don't retry downloads when disk is full

There was already a check that prevented protocol-level fallback in this
situation, but retries within a specific protocol will still happen.
This makes it take a long time for the pull to finally error out.

This fixes slowness in TestDaemonNoSpaceleftOnDeviceError, which used to
take a long time due to the backoff between retry attempts:

PASS: docker_cli_daemon_test.go:1868: DockerDaemonSuite.TestDaemonNoSpaceleftOnDeviceError	5.882s

Signed-off-by: Aaron Lehmann <aaron.lehmann@docker.com>

Commit:
Vendor updated distribution package

Another day, another revendor.

This revision of distribution is more tolerant of incorrect Content-Type
headers when fetching manifests.

Fixes #19526

Signed-off-by: Aaron Lehmann <aaron.lehmann@docker.com>

aboch committed:
Reject multiple networks on container creation

Signed-off-by: Alessandro Boch <aboch@docker.com>

aidanhs committed:
Permit OPTIONS request against any url, fixes #19398

Signed-off-by: Aidan Hobson Sayers <aidanhs@cantab.net>

cpuguy83 committed:
On container rm, don't remove named mountpoints

This makes it so when calling `docker run --rm`, or `docker rm -v`, only
volumes specified without a name, e.g. `docker run -v /foo` instead of
`docker run -v awesome:/foo` are removed.

Note that all volumes are named, some are named by the user, some get a
generated name. This is specifically about how the volume was specified
on `run`, assuming that if the user specified it with a name they expect
it to persist after the container is cleaned up.

Signed-off-by: Brian Goff <cpuguy83@gmail.com>

Commit:
Vendor in libnetwork v0.6.0-rc2

Signed-off-by: Santhosh Manohar <santhosh@docker.com>

Commit:
IT for service/network name with '.', corrected libnetwork flag for DNS

Signed-off-by: Santhosh Manohar <santhosh@docker.com>

polonskiy committed:
fixed typo

Signed-off-by: Anton Polonskiy <anton.polonskiy@gmail.com>

cyphar committed:
vendor: *: update libcontainer

This fixes a security vulnerability in Docker, which can cause a DoS
under certain circumstances. This is from the hotfix branch, so the
vendored commit is actually bf899fef451956be4abd63de6d6141d9f9096a02 in
runc master.

Signed-off-by: Aleksa Sarai <asarai@suse.com>

tianon committed:
Add a note to /etc/default/docker noting that it does not apply to systemd

Signed-off-by: Andrew "Tianon" Page <admwiggin@gmail.com>

Commit:
Add some helper text for magical ADD

Closes: #15777

Signed-off-by: Doug Davis <dug@us.ibm.com>

calavera committed:
Make TLSOptions and LogConfig embedded structs.

That way the configuration file becomes flag, without extra keys.

Signed-off-by: David Calavera <david.calavera@gmail.com>

Djelibeybi committed:
Updated Oracle Linux install documentation to be more accurate.

Signed-off-by: Avi Miller <avi.miller@oracle.com>

tianon committed:
Add pkg-config to our Debian build environment

This is used in `hack/make.sh` for detecting various dependencies such as `libsystemd-journal` -- without this, our packages don't support pulling logs back out of journald. :cry:

Signed-off-by: Andrew "Tianon" Page <admwiggin@gmail.com>

nalind committed:
Add pkg-config to our RPM build environment

While hack/make.sh checks for systemd headers using pkg-config, we
forgot to ensure that they were there in the images that we use for
building binaries for RPM-based distributions.  Add the right packages
to the generate.sh that we use for them, and update the copies of the
generated files that we carry in the source tree.

Notes: Fedora, CentOS, and Oracle Linux put the pkg-config command in
the "pkgconfig" package, while OpenSUSE calls the package "pkg-config".
The systemd-devel package, like systemd, is not in Oracle Linux 6.

Signed-off-by: Nalin Dahyabhai <nalin@redhat.com>

hqhq committed:
Verify cgroup-parent name for systemd cgroup

Fixes: #17126

Signed-off-by: Qiang Huang <h.huangqiang@huawei.com>

Commit:
Creating Engine specific menu
Fixing the links
Updating with Seb's comments
Adding weight
Fixing the engine aliases
Updating after Arun pushed
Removing empty file

Signed-off-by: Mary Anthony <mary@docker.com>

r4j4h committed:
Correcting `overlay` -> `bridge` driver in run.md

Correcting `overlay` -> `bridge` driver in run.md to match the preceding paragraph.

Signed-off-by: Jasmine Hegman <jasmine@jhegman.com>

thaJeztah committed:
Fix remove API order in menu

Signed-off-by: Sebastiaan van Stijn <github@gone.nl>

albers committed:
bash completion for `docker cp --follow-link`

Signed-off-by: Harald Albers <github@albersweb.de>

albers committed:
bash completion for `docker ps --filter status=dead`

Signed-off-by: Harald Albers <github@albersweb.de>

albers committed:
bash completion for `docker images -f dangling=false`

Signed-off-by: Harald Albers <github@albersweb.de>

albers committed:
bash completion for `--cluster-store-opt kv.path discovery.{heartbeat,ttl}`

Signed-off-by: Harald Albers <github@albersweb.de>

albers committed:
bash completion for syslog over TLS log driver

Signed-off-by: Harald Albers <github@albersweb.de>

albers committed:
remove zsh completion for `docker tag -f`

Signed-off-by: Harald Albers <github@albersweb.de>

albers committed:
fix minor bash completion issue on OSX (compopt)

Signed-off-by: Harald Albers <github@albersweb.de>

nishanttotla committed:
Adding SystemStatus field for /info endpoint

Signed-off-by: Nishant Totla <nishanttotla@gmail.com>

sdurrheimer committed:
Add zsh completion for 'docker cp -L --follow-link'

Signed-off-by: Steve Durrheimer <s.durrheimer@gmail.com>

sdurrheimer committed:
Add zsh completion for new 'docker daemon --cluster-store-opt discovery.heartbeat discovery.ttl kv.path' options

Signed-off-by: Steve Durrheimer <s.durrheimer@gmail.com>

sdurrheimer committed:
Add zsh completion for new 'docker daemon --log-opt syslog-tls-ca-cert syslog-tls-cert syslog-tls-key syslog-tls-skip-verify' options

Signed-off-by: Steve Durrheimer <s.durrheimer@gmail.com>

WeiZhang555 committed:
Forbid exec a restarting container

Currently if we exec a restarting container, client will fail silently,
and daemon will print error that container can't be found which is not a
very meaningful prompt to user.

This commit will stop user from exec a restarting container and gives
more explicit error message.

Signed-off-by: Zhang Wei <zhangwei555@huawei.com>

aboch committed:
Save endpoint config only if endpoint creation succeeds

- Currently it is being save upfront...

Signed-off-by: Alessandro Boch <aboch@docker.com>

Commit:
Fix watching a released transfer

Things could go wrong if Watch was called after the last existing
watcher was released. The call to Watch would succeed even though it was
not really adding a watcher, and the corresponding call to Release would
close hasWatchers a second time.

The fix for this is twofold:

1. We allow transfers to gain new watchers after the watcher count has
touched zero. This means that the channel returned by Released should
not be closed until all watchers have been released AND the transfer is
no longer tracked by the transfer manager, meaning it won't be possible
for additional calls to Watch to race with closing the channel returned
by Released.

The Transfer interface has a new method called Close so the transfer can
know when the transfer manager no longer references it.

Remove the Cancel method. It's not used and should not be exported.

2. Even though (1) makes it possible to add watchers after all the
previous watchers have been released, we want to avoid doing this in
practice. A transfer that has had all its watchers released is in the
process of being cancelled, and attaching to one of these will never be
the correct behavior. Add a check if a watcher is attaching to a
cancelled transfer.  In this case, wait for the transfer to be removed
from the map and try again. This will ensure correct behavior when a
watcher tries to attach during the race window.

Either (1) or (2) should be sufficient to fix the race involved here,
but the combination is the most correct approach. (1) fixes the
low-level plumbing to be resilient to the race condition, and (2) avoids
using it in a racy way.

Fixes #19606

Signed-off-by: Aaron Lehmann <aaron.lehmann@docker.com>

calavera committed:
Make volume dangling filter return only used volumes with `dangling=false`.

Signed-off-by: David Calavera <david.calavera@gmail.com>

tonistiigi committed:
Fix error message in container creation

Error message was different if image was specified with the full ID.

Fixes #19652

Signed-off-by: Tonis Tiigi <tonistiigi@gmail.com>

cpuguy83 committed:
Fix removing mountpoints on container rm fail

Ensure that the the container's mountpoints are cleaned up if the
container is force removed.

Signed-off-by: Brian Goff <cpuguy83@gmail.com>

mavenugo committed:
Vendor libnetwork v0.6.0-rc3

- Fixes docker/docker#19576
- Fixed embedded DNS to listen in TCP as well
- Fixed a race-condition in IPAM to choose non-overlapping subnet for concurrent requests

Signed-off-by: Madhu Venugopal <madhu@docker.com>

calavera committed:
Allow network configuration via daemon config file.

Signed-off-by: David Calavera <david.calavera@gmail.com>

crosbymichael committed:
Move tar copy-up for tmpfs mounts

We cannot rely on the tar command for this type of operation because tar
versions, flags, and functionality can very from distro to distro.
Since this is in the container execution path it is not safe to have
this as a dependency from dockers POV where the user cannot change the
fact that docker is adding these pre and post mount commands.

Signed-off-by: Michael Crosby <crosbymichael@gmail.com>

tianon committed:
Change the way we install from backports in the deb builder (to force deps too)

Also, add "libsystemd-journal-dev" to the explicit list (which is what prompted the change in how we install).

Signed-off-by: Andrew "Tianon" Page <admwiggin@gmail.com>

Commit:
Remove temporary layer download file on error

Currently, the temporary file storing downloaded layer data is only
removed after a successful download or a digest verification error. A
transport-level error does not cause it to be removed. This is a
regression from 1.9 that could cause disk usage to grow until the Docker
daemon is restarted.

Signed-off-by: Aaron Lehmann <aaron.lehmann@docker.com>

mavenugo committed:
Vendor libnetwork v0.6.0-rc4

- Add Endpoints() API to Sandbox interface
- Fixes #19677

Signed-off-by: Madhu Venugopal <madhu@docker.com>

mavenugo committed:
Move port-mapping ownership closer to Sandbox (from Endpoint)

https://github.com/docker/libnetwork/pull/810 provides the more complete
solution for moving the Port-mapping ownership away from endpoint and
into Sandbox. But, this PR makes the best use of existing libnetwork
design and get a step closer to the gaol.

Signed-off-by: Madhu Venugopal <madhu@docker.com>

cpuguy83 committed:
Use imageStore.Children instead of whole the Map

daemon cache was getting the whole image map and then iterating through
it to find children. This information is already stored in the image
store.

Prior to this change building the docker repo with a full cache took 30
seconds.
After it takes between 15 seconds or less (As low as 9 seconds).
This is an improvement on docker 1.9.1 which hovered around 17 seconds.

Signed-off-by: Brian Goff <cpuguy83@gmail.com>

pwnall committed:
Add missing words in selectadriver.md

Signed-off-by: Victor Costan <costan@gmail.com>

bboreham committed:
Improve wording about re-assigning IP addresses

Signed-off-by: Bryan Boreham <bjboreham@gmail.com>

albers committed:
bash completion for `docker daemon --userns-remap`

Signed-off-by: Harald Albers <github@albersweb.de>

calavera committed:
Respond with 401 when there is an unauthorized error from the registry.

Signed-off-by: David Calavera <david.calavera@gmail.com>

calavera committed:
Fix bash completion for `docker volume ls --dangling=false`.

Signed-off-by: David Calavera <david.calavera@gmail.com>

mrjana committed:
Vendoring libnetwork v0.6.0-rc5

  - Cleanup stale overlay sandboxes

Fixes #19694

Signed-off-by: Jana Radhakrishnan <mrjana@docker.com>

thaJeztah committed:
Update notary to 1.10-3 in all Dockerfiles

Signed-off-by: Sebastiaan van Stijn <github@gone.nl>

jessfraz committed:
way better apparmor docs

Signed-off-by: Jessica Frazelle <acidburn@docker.com>

Commit:
Set idle timeouts for HTTP reads and writes in communications with the registry

Otherwise, some operations can get stuck indefinitely when the remote
side is unresponsive.

Fixes #12823

Signed-off-by: Aaron Lehmann <aaron.lehmann@docker.com>

calavera committed:
Remove cluster storage advertise from reload.

Because libnetwork won't really send container information to the new
storage anyways.

Signed-off-by: David Calavera <david.calavera@gmail.com>

tkopczynski committed:
FAQ: add docker daemon unavailable note

Signed-off-by: Tomasz Kopczynski <tomek@kopczynski.net.pl>

cpuguy83 committed:
Add note about mount propagation on systemd

Signed-off-by: Brian Goff <cpuguy83@gmail.com>

cyli committed:
Re-vendor notary, as well as change jfrazelle/go to docker/go.

Signed-off-by: cyli <cyli@twistedmatrix.com>

crosbymichael committed:
Update libcontainer to 3d8a20bb772defc28c355534d83

Fixes #14203

This bump fixes the issue of having the container's pipes connection
reset by peer because of using the json.Encoder and having a \n added to
the output.

Signed-off-by: Michael Crosby <crosbymichael@gmail.com>

thaJeztah committed:
Fix docs for tmpfs (pr 19688)

Underlying files are no longer copied to the tmpfs.

Signed-off-by: Sebastiaan van Stijn <github@gone.nl>

cyli committed:
Update integration tests with new error messages, and to use different repos per test.

This way we won't encounter any problems with one test using cached data from a different
test.

Signed-off-by: cyli <cyli@twistedmatrix.com>

albers committed:
Improve bash completion for `docker volume ls -f dangling`

Signed-off-by: Harald Albers <github@albersweb.de>

patricksuo committed:
fix dead lock in volume store dereference

Signed-off-by: Pei Su <sillyousu@gmail.com>

lukeck commented:
Changelog entry "New registrytoken pass-thru token type for AuthConfig" incorrectly references #17481. Should be #17741.

cyli committed:
Include a new version of notary with less verbose INFO+ logging

Signed-off-by: cyli <cyli@twistedmatrix.com>

Commit:
Update Instructions for Docker Remote API access

Update the location of cert.pem and key.pem to use within the
~/.docker folder
Resolves: #18778

Signed-off-by: Aditi Rajagopal <arajagopal@us.ibm.com>

cyli commented:
Apologies, could we also include the following changelog entry:  (https://github.com/docker/docker/pull/18887) "Docker Content Trust now supports the ability to read and write user delegations. This is an optional, opt-in feature that requires the explicit use of the Notary command-line utility in order to be enabled. Enabling delegation support in a specific repository will break the ability of Docker 1.9 and 1.8 to pull from that repository, if content trust is enabled."

calavera committed:
Always prompt for a password when asking for credentials.

There is a weird behavior where we don't ask for a password
when the user you type in the prompt is the same you have configured
in the config file.

This is the source of many frustrations and also a bug.
If the authentication with a registry fails because the password
is incorrect, we won't ask for the password again with the current logic.

With this change, we also stop calling `CmdLogin` directly when
authentication fails. We don't need to parse flags from the cli or
setting up input destriptiors again, like the current behavior is doing.

Signed-off-by: David Calavera <david.calavera@gmail.com>

thaJeztah commented:
ping @burke wondering if you've been giving docker 1.10 a spin already; welcoming feedback :+1:

Commit:
Add doc page for breaking changes

Signed-off-by: Arnaud Porterie <arnaud.porterie@docker.com>

Commit:
Change container name to id as actual results

Signed-off-by: Wen Cheng Ma <wenchma@cn.ibm.com>

jessfraz committed:
update bash completions for push and pull

Signed-off-by: Jessica Frazelle <acidburn@docker.com>

diogomonica commented on CHANGELOG.md:
```diff
@@ -5,6 +5,133 @@ information on the list of deprecated flags and APIs please have a look at
+* Add support for the new [manifest format ("schema2")](https://github.com/docker/distribution/blob/master/docs/spec/manifest-v2-2.md) [#18785](https://github.com/docker/docker/pull/18785)
+* Lots of improvements for push and pull: performance++, retries on failed downloads, cancelling on client disconnect [#18353](https://github.com/docker/docker/pull/18353), [#18418](https://github.com/docker/docker/pull/18418), [#19109](https://github.com/docker/docker/pull/19109), [#18353](https://github.com/docker/docker/pull/18353)
+* Limit v1 protocol fallbacks [#18590](https://github.com/docker/docker/pull/18590)
+* New version of notary with client-side pkcs11 support [#17937](https://github.com/docker/docker/pull/17937)
```
This should be replaced with: "New version of notary with delegations support"

thaJeztah replied:
@diogomonica in addition to pkcs11 support? Because that PR looks to be in 1.10 (previously it was only in experimental, I think?)

diogomonica replied:
@thaJeztah that wasn't intentional from our part, but the code seems to have been solid over the past three months, so it might be ok. I think the majority of the issues around putting it in master had to do with the inability of doing static binaries (we are using cgo).

Commit:
Fix panic on network timeout during push

`Upload` already closes the reader returned by `compress` and the
progressreader passed into it, before returning. But even so, the
io.Copy inside compress' goroutine needs to attempt a read from the
progressreader to notice that it's closed, and this read has a side
effect of outputting a progress message. If this happens after `Upload`
returns, it can result in a write to a closed channel. Change `compress`
to return a channel that allows the caller to wait for its goroutine to
finish before freeing any resources connected to the reader that was
passed to it.

Signed-off-by: Aaron Lehmann <aaron.lehmann@docker.com>

shin- committed:
Update example request for container update in docs.

Signed-off-by: Joffrey F <joffrey@docker.com>

aboch committed:
Vendoring libnetwork 0.6.0-rc6

Signed-off-by: Alessandro Boch <aboch@docker.com>

mavenugo committed:
Docs cleanup for networking features added in v1.10

Signed-off-by: Madhu Venugopal <madhu@docker.com>

prayagverma committed:
Fix typo

Signed-off-by: Prayag Verma <prayag.verma@gmail.com>

ZhuWeiyang committed:
Fix upgrade command

Signed-off-by: Weiyang Zhu <cnresonant@gmail.com>

Qiaohai8866 committed:
change 'host:port' to `host:port`

Signed-off-by: Gang Qiao <qiaohai8866@gmail.com>

Commit:
Added link to Quobyte's volume plugin.

Signed-off-by: Felix Hupfeld <felix@quobyte.com>

vdemeester committed:
Add docker-volume-ipfs plugin to the list.

Signed-off-by: Vincent Demeester <vincent@sbr.pm>

thaJeztah committed:
Add note about legacy links

Signed-off-by: Sebastiaan van Stijn <github@gone.nl>

Commit:
Enable cross-platforms login to Registry

Use a daemon-defined Registry URL for `docker login`. This allows a
Windows client interacting with a Linux daemon to properly use the
default Registry endpoint instead of the Windows specific one.

Signed-off-by: Arnaud Porterie <arnaud.porterie@docker.com>

cpuguy83 committed:
handle debug mode for clients

Signed-off-by: Brian Goff <cpuguy83@gmail.com>

joelwurtz committed:
Update docker-php library from stage1

Signed-off-by: Joel Wurtz <joel.wurtz@gmail.com>

aboch committed:
Vendoring libnetwork 0.6.0-rc7

Signed-off-by: Alessandro Boch <aboch@docker.com>

Commit:
Skip Close()-ing stdin on Darwin. The Return.

This was accidentally removed in https://github.com/docker/docker/pull/16289
Now adding it back.

Signed-off-by: Tibor Vass <tibor@docker.com>

Commit:
Correct old virtual size

In new content addressable model, image no longer
have virtual size column, it is now 'size'. So we
need to update related docs about them.

Signed-off-by: Kai Qiang Wu(Kennan) <wkqwu@cn.ibm.com>

thaJeztah committed:
Fix memory-swap description in older API versions

A value of -1 disables the *limit* so enables unlimited swap

Signed-off-by: Sebastiaan van Stijn <github@gone.nl>

Commit:
Changing menu label from Engine > Docker Engine

Signed-off-by: Mary Anthony <mary@docker.com>

Commit:
Delegation is not experimental

Change wording that described user delegation as an experimental
feature.

Signed-off-by: Arnaud Porterie <arnaud.porterie@docker.com>

Commit:
vendor engine-api 0.2.3 to add SystemStatus in Info type

Signed-off-by: Tibor Vass <tibor@docker.com>

calavera committed:
Make sure flat options are not parsed as config structures.

Signed-off-by: David Calavera <david.calavera@gmail.com>

jhowardmsft commented:
I think (digging) that windowsTP4 CI is failing because the PR to merge dockerfile.Windows isn't in the cherry-pick list. https://github.com/docker/docker/pull/19529

vieux committed:
fix error message on version too low

Signed-off-by: Victor Vieux <vieux@docker.com>

bfirsh committed:
Add 1.10 migration docs

Originally from https://blog.docker.com/2016/01/docker-1-10-rc/

Signed-off-by: Ben Firshman <ben@firshman.co.uk>

aboch committed:
Store endpoint config on network connect to a stopped container

Signed-off-by: Alessandro Boch <aboch@docker.com>

coolljt0725 committed:
Check nil before set resource.OomKillDisable

Signed-off-by: Lei Jitang <leijitang@huawei.com>

marsmensch committed:
Fix formatting in breaking changes doc

Signed-off-by: Florian Maier <contact@marsmenschen.com>

Commit:
Revert "Set idle timeouts for HTTP reads and writes in communications with the registry"

This reverts commit 84b2162c1a15256ac09396ad0d75686ea468f40c.

The intent of this commit was to set an idle timeout on a HTTP
connection. If a read took more than 60 seconds to complete, or a write
took more than 60 seconds to complete, the connection would be
considered dead.

This doesn't work properly, because the HTTP internals apparently read
from the connection concurrently while writing. An upload that doesn't
complete in 60 seconds leads to a timeout.

Fixes #19967

Signed-off-by: Aaron Lehmann <aaron.lehmann@docker.com>

Commit:
Enable cross-platforms logout from Registry

Signed-off-by: Arnaud Porterie <arnaud.porterie@docker.com>

Commit:
Remove unnecessary call to /info

Avoid using the `/info` endpoint in the `login` and `logout` workflows
when the Registry endpoint is overriden by the user through the command
line.

Signed-off-by: Arnaud Porterie <arnaud.porterie@docker.com>

Commit:
Introduce a client-side version of resolveAuthConfig

This is similar to the version in the registry package, but uses the
daemon's default index (as opposed to the default for the client's
platform) if using the "official index".

Signed-off-by: Aaron Lehmann <aaron.lehmann@docker.com>

thaJeztah committed:
dont try to install journald driver on wheezy

Signed-off-by: Sebastiaan van Stijn <github@gone.nl>
Signed-off-by: Jessica Frazelle <acidburn@docker.com>

jfrazelle commented:
LGTM

cpuguy83 commented:
LGTM

thaJeztah commented:
LGTM!
